### PR TITLE
ACI-CNI 5.2.3.4 support

### PR DESCRIPTION
**Pull request by LavanyaVaddavalli**

### CODE REVIEW

**Comment by snaiksat** [comment]
@kinarashah  @snasovich

File: rke/templates/templates.go
Comment (snasovich) [inline rke/templates/templates.go]: I believe we should use `v1.22.13-rancher1-2` instead of `v1.22.13-rancher1-1` in these 2 lines as for `v1.22.13-rancher1-1` (already existing version) `aciv523` template should be used, but the current conditions will result in `aciv5233`.

File: rke/templates/templates.go
Comment (snasovich) [inline rke/templates/templates.go]: Same as above, I believe `v1.23.10-rancher1-2` should be used instead.

File: rke/templates/templates.go
Comment (snasovich) [inline rke/templates/templates.go]: Same as above, `v1.24.4-rancher1-2` is the first version where `aciv5233` should apply to.

File: rke/templates/templates.go
Comment (snasovich) [inline rke/templates/templates.go]: Nit, but we want variables and values to align, see other blocks like this.

File: rke/templates/templates.go
Comment (snasovich) [inline rke/templates/templates.go]: Nit, but we want variables and values to align, see other blocks like this.

**Review by snasovich — CHANGES_REQUESTED** [review]
Posted some specific comments inline.

However, we're introducing new versions for 2.7.0 release as per https://github.com/rancher/kontainer-driver-metadata/pull/967 so we want updated ACI pluggin for these newly introduced versions. We cannot have what will be newer versions (like `v1.22.15-rancher1-1`) have older version of ACI plugin than what would be older version (like `v1.22.13-rancher1-2`).

It's best to wait for the referenced PR to be merged and update ACI specific image versions as well as introduce new template and template conditions.

**Comment by LavanyaVaddavalli** [comment]
@snasovich addressed all comments in commit d4d4d56db8839fea2e11f7318c3708d21b357a8d

File: rke/templates/templates.go
Comment (LavanyaVaddavalli) [reply rke/templates/templates.go]: addressed in d4d4d56db8839fea2e11f7318c3708d21b357a8d

File: rke/templates/templates.go
Comment (LavanyaVaddavalli) [reply rke/templates/templates.go]: addressed in d4d4d56db8839fea2e11f7318c3708d21b357a8d

File: rke/templates/templates.go
Comment (LavanyaVaddavalli) [reply rke/templates/templates.go]: addressed in d4d4d56db8839fea2e11f7318c3708d21b357a8d

File: rke/templates/templates.go
Comment (LavanyaVaddavalli) [reply rke/templates/templates.go]: addressed in d4d4d56db8839fea2e11f7318c3708d21b357a8d

File: rke/templates/templates.go
Comment (LavanyaVaddavalli) [reply rke/templates/templates.go]: addressed in d4d4d56db8839fea2e11f7318c3708d21b357a8d

**Comment by snaiksat** [comment]
Thanks @snasovich , we might need to do another update on this patch with regards to the image tags. So kindly bear with us while we do that.

**Comment by snasovich** [comment]
@LavanyaVaddavalli @snaiksat , thank you for the updates.
Please note that https://github.com/rancher/kontainer-driver-metadata/pull/967 was merged a week ago, so this PR is now ready to be updated as per this comment: https://github.com/rancher/kontainer-driver-metadata/pull/968#pullrequestreview-1119065782

**Comment by siva-muni** [comment]
@kinarashah @snasovich, we updated the image tags and completed code changes. Can you please review this PR

**Comment by snasovich** [comment]
@snaiksat @siva-muni @LavanyaVaddavalli , I know it must feel like hitting a fast-moving target, but we're releasing 2.7.0 today and then the next release of KDM is planned for 11/23 (next Wednesday) to introduce k8s November patches that were released last week.

The PR for 2.7.x release line is https://github.com/rancher/kontainer-driver-metadata/pull/999 and should be merged today. Notice the target branch is [dev-v2.7-1122-patches](https://github.com/rancher/kontainer-driver-metadata/tree/dev-v2.7-1122-patches) and not `dev-v2.7` - the former is the branch will be releasing the patch from.

So, if we want to have ACI 5.2.3.4 support in this 11/23 upcoming release, this will have to be merged by end of this week. For that:
* This PR (or a new one if it's easier to just start fresh) will need to be done against `dev-v2.7-1122-patches` branch and be rebased to include changes from https://github.com/rancher/kontainer-driver-metadata/pull/999 (after it's merged - which should be later today)
* Instead of introducing new versions (like `v1.24.6-rancher1-2`), we want to simply update ACI images referenced by versions newly introduced by https://github.com/rancher/kontainer-driver-metadata/pull/999 - so `v1.22.16-rancher1-1`, `v1.23.14-rancher1-1`, and `v1.24.8-rancher1-1`. Template changes and version ranges will also need to be updated so that `AciTemplateV5234` template is only used starting with these newly introduced versions.
* Another PR making the same changes against `dev-v2.6-1122-patches` branches should also be opened. PR https://github.com/rancher/kontainer-driver-metadata/pull/1000 is what will introduce November k8s versions for that branch  (should also be merged today).

If the above can be made available for review in the next couple days, we have a good shot at including updated ACI version in 11/23 release. Otherwise, the next opportunity to release it will be with December 2022 patches late in December (it will be the same process where there will be PRs introducing these versions and we will want ACI versions bumped on them).

Hope this is helpful, let me know how you prefer to proceed here.

**Comment by snaiksat** [comment]
Thanks @snasovich  for providing us with the options on how to proceed. We will are discussing internally but we will try to meet timeline you suggested.

**Comment by siva-muni** [comment]
Thanks @snasovich for the suggestion. As per your suggestion, we raised two PRs one is  [https://github.com/rancher/kontainer-driver-metadata/pull/1017](https://github.com/rancher/kontainer-driver-metadata/pull/1017) against  "dev-v2.7-1122-patches" and  another is [https://github.com/rancher/kontainer-driver-metadata/pull/1016](https://github.com/rancher/kontainer-driver-metadata/pull/1016)  against to "dev-v2.6-1122-patches".  Please review The PRs.

**Comment by snasovich** [comment]
@siva-muni , thank you! I've reviewed referenced PRs and they LGTM. Need couple more people to take a look into them and hopefully we can merge this week.
Closing this one as it's effectively replaced by the other 2 PRs.

**Pull request closed without merging**
